### DISCUSSION TITLE
Negative patterns in `consult-ripgrep`

### DISCUSSION THREAD

**Discussion by NightMachinery** [minad/consult#948] [Q&A]
I like to have some negative patterns when using `consult-ripgrep`, e.g., `#ML !#AI` (exclude matches with `#AI`). I use `orderless` which compiles these negative patterns and they work with `consult-line`, but not with `consult-ripgrep`.

PS: Is `ugrep` supported? `ugrep --bool` is the best regexp search mode, and it doesn't need any hacks like `orderless`.

**Comment by minad** [comment]
`#ml#!ai` should work. The pattern behind the second # is handled by Orderless. Ugrep is not supported but I think there have been attempts for a `consult-ugrep` on the issue tracker. I don't want to support it officially here though, given that so many similar search utilities exist.

**Reply by NightMachinery** [reply]
Can you elaborate on how that works?

How can I write this in consult?
```
IDEA\\|@idea !@ideas !@idea/rejected !@idea/bad !@idea/presented !@idea/costly !@idea/small
```
